app/views/ProfileView/index.tsx

```diff
@@ -96,138 +132,95 @@ class ProfileView extends React.Component<IProfileViewProps, IProfileViewState>
 		navigation.setOptions(options);
 	};
 
-	constructor(props: IProfileViewProps) {
-		super(props);
-		this.setHeader();
-	}
-
-	state: IProfileViewState = {
-		saving: false,
-		name: '',
-		username: '',
-		email: '',
-		bio: '',
-		nickname: '',
-		newPassword: '',
-		currentPassword: '',
-		customFields: {},
-		twoFactorCode: null
-	};
-
-	componentDidMount() {
-		this.focusListener = this.props.navigation.addListener('focus', () => {
-			this.init();
-		});
-	}
-
-	componentWillUnmount() {
-		this.focusListener();
-	}
-
-	init = (user?: IUser) => {
-		const { user: userProps } = this.props;
-		const { name, username, emails, customFields, bio, nickname } = user || userProps;
-
-		this.setState({
-			name: name as string,
-			username,
-			email: emails ? emails[0].address : null,
-			newPassword: null,
-			currentPassword: null,
-			customFields: customFields || {},
-			bio,
-			nickname
-		});
-	};
-
-	formIsChanged = () => {
-		const { name, username, email, newPassword, customFields, bio, nickname } = this.state;
-		const { user } = this.props;
+	const formIsChanged = React.useCallback(() => {
 		let customFieldsChanged = false;
 
-		const customFieldsKeys = Object.keys(customFields);
+		const customFieldsKeys = Object.keys(state.customFields);
 		if (customFieldsKeys.length) {
 			customFieldsKeys.forEach(key => {
-				if (!user.customFields || user.customFields[key] !== customFields[key]) {
+				if (!user.customFields || user.customFields[key] !== state.customFields[key]) {
 					customFieldsChanged = true;
 				}
 			});
 		}
 
 		return !(
-			user.name === name &&
-			user.username === username &&
-			user.bio === bio &&
-			user.nickname === nickname &&
-			!newPassword &&
+			user.name === state.name &&
```

Comment:
I don't know if you personally changed it or used some tool to make this change, but just this block here "kills" the migration, using a simple state for this entire flow will make the performance of this screen much worse.
The correct option would be useReduce or a form lib.

Reply:
#5544  This PR will solve the problem. This PR is for integrating `react-hook-form`. Let me know if any other files, like `RegisterView` need to be changed from class to function components.